src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs

```diff
@@ -63,20 +63,15 @@ public class WorkitemOrchestrator : IWorkitemOrchestrator
 
                 IReadOnlyCollection<QueryTag> queryTags = await _workitemQueryTagService.GetQueryTagsAsync(cancellationToken).ConfigureAwait(false);
 
-                long workitemKey = await _indexWorkitemStore
+                identifier = await _indexWorkitemStore
                     .BeginAddWorkitemAsync(partitionKey, dataset, queryTags, cancellationToken)
                     .ConfigureAwait(false);
 
-                identifier = new WorkitemInstanceIdentifier(
-                    dataset.GetSingleValueOrDefault(DicomTag.SOPInstanceUID, string.Empty),
-                    workitemKey,
-                    partitionKey);
-
                 // We have successfully created the index, store the file.
                 await StoreWorkitemBlobAsync(identifier, dataset, cancellationToken)
                     .ConfigureAwait(false);
 
-                await _indexWorkitemStore.EndAddWorkitemAsync(partitionKey, workitemKey, cancellationToken);
+                await _indexWorkitemStore.EndAddWorkitemAsync(partitionKey, identifier.WorkitemKey, cancellationToken);
```

Comment:
We can include watermark in the TryDeleteWorkitemAsync method

Reply:
Done.